mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp

```diff
@@ -171,6 +171,50 @@ static DiagnosedSilenceableFailure unpackSingleIndexResultPayloadOperations(
   return DiagnosedSilenceableFailure::success();
 }
 
+static DiagnosedSilenceableFailure reifyMixedParamAndHandleResults(
+    TransformState &state, TransformOpInterface &transformOp,
+    const SmallVectorImpl<OpFoldResult> &mixedResults,
+    SmallVectorImpl<int64_t> &reified) {
+  for (OpFoldResult paramOrHandle : mixedResults) {
+    if (isa<Attribute>(paramOrHandle)) {
+      reified.push_back(
+          cast<IntegerAttr>(paramOrHandle.get<Attribute>()).getInt());
+      continue;
+    } else if (isa<Value>(paramOrHandle) &&
+               isa<ParamType>(paramOrHandle.get<Value>().getType())) {
+      ArrayRef<Attribute> params = state.getParams(paramOrHandle.get<Value>());
+      if (params.size() != 1)
+        return transformOp.emitDefiniteFailure() << "expected a single param";
+      reified.push_back(
+          cast<IntegerAttr>(params.front()).getValue().getSExtValue());
+      continue;
+    }
+
+    auto paramOrHandlePayloads =
+        state.getPayloadOps(paramOrHandle.get<Value>());
```

Comment:
On the other hand, we haven't checked that it isn't a value handle.

Reply:
hmm, not quite sure what you mean

Reply:
We have (operation) handles and value handles. The fact the something is not a parameter doesn't necessarily make it an operation handle that `getPayloadOps` expects, it may be a value handle. So this will have to check, and react accordingly, if `paramOrHandle` is a value handle. Or we can disallow value handles in the type constraint of the op that uses this and add an assertion here.

Reply:
so i guess i was confused by your other comment https://github.com/llvm/llvm-project/pull/90755#discussion_r1587706648. so i do indeed need to check if value for both then?